src/taco-dependency-installer/installers/androidPackagesInstaller.ts

```diff
@@ -77,26 +82,25 @@ class AndroidPackagesInstaller extends InstallerBase {
         }
     }
 
-    private getAvailableAndroidPackages(): Q.Promise<any> {
-        var deferred: Q.Deferred<any> = Q.defer();
+    private executeAndroidCommand(name: string, args: string[]): IExecuteCommandResult {
```

Comment:
I think it would be more intuitive to just return a promise resolved with stdout... I think it's slightly confusing to have to use this method by chaining to `result.output` like such:

```typescript
var cp: IExecuteCommandResult = this.executeAndroidCommand("getAvailableAndroidPackages", args);
return cp.output.then((output: string) => {
```

I understand that you had to do it this way for the `installAndroidPackages` method to be able to listen for the "[y/n]:" prompt. I understand that we want `executeAndroidCommand` to be as generic as possible, but I still feel it would make sense to move the "[y/n]:" prompt logic to `executeAndroidCommand`. This way, consumers that want to invoke an Android command don't need to worry about any prompt, and we get to change the signature to just return a promise for stdout. We can then use this method like so:

```typescript
this.executeAndroidCommand("getAvailableAndroidPackages", args).then((output: string) => {
...
});
```

which I believe is much more intuitive. But it's just my opinion, feel free to disagree!